src/sidebar/store/test/use-store-test.js

```diff
@@ -16,90 +18,262 @@ const reducer = (state = initialState, action) => {
+      });
+      assert.equal(renderCount, originalRenderCount);
+    });
+
+    it('warns if the callback always returns a different value', () => {
+      const warnOnce = sinon.stub();
+      $imports.$mock({
+        '../../shared/warn-once': warnOnce,
+      });
+      const BuggyComponent = () => {
+        // The result of the callback is an object with an `aValue` property
+        // which is a new array every time. This causes unnecessary re-renders.
+        useStore(() => ({ aValue: [] }));
+        return null;
+      };
+      mount(<BuggyComponent />);
+      assert.called(warnOnce);
+      assert.match(warnOnce.firstCall.args[0], /changes every time/);
+    });
+
+    it('unsubscribes when the component is unmounted', () => {
+      const unsubscribe = sinon.stub();
+      testStore.subscribe = sinon.stub().returns(unsubscribe);
+
+      const wrapper = mount(<TestComponent />);
+
+      assert.calledOnce(testStore.subscribe);
+      wrapper.unmount();
+      assert.calledOnce(unsubscribe);
     });
-    const BuggyComponent = () => {
-      // The result of the callback is an object with an `aValue` property
-      // which is a new array every time. This causes unnecessary re-renders.
-      useStore(() => ({ aValue: [] }));
-      return null;
-    };
-    mount(<BuggyComponent />);
-    assert.called(warnOnce);
-    assert.match(warnOnce.firstCall.args[0], /changes every time/);
   });
 
-  it('unsubscribes when the component is unmounted', () => {
-    const unsubscribe = sinon.stub();
-    testStore.subscribe = sinon.stub().returns(unsubscribe);
+  describe('useStoreProxy', () => {
+    let store;
+    let renderCount;
+
+    beforeEach(() => {
+      renderCount = 0;
+      store = createStore([thingsModule]);
+
+      store.addThing('foo');
+      store.addThing('bar');
+
+      $imports.$mock({
+        '../util/service-context': {
+          useService: name => (name === 'store' ? store : null),
+        },
+      });
+    });
+
+    function renderTestComponent() {
+      let proxy;
+
+      const TestComponent = () => {
+        ++renderCount;
+        proxy = useStoreProxy();
+
+        return <div>{proxy.thingCount()}</div>;
+      };
+
+      const wrapper = mount(<TestComponent />);
+      return { proxy, wrapper };
+    }
+
+    it('returns a proxy for the store', () => {
+      const addThingSpy = sinon.spy(store, 'addThing');
 
-    const wrapper = mount(<TestComponent />);
+      const { proxy } = renderTestComponent();
 
-    assert.calledOnce(testStore.subscribe);
-    wrapper.unmount();
-    assert.calledOnce(unsubscribe);
+      assert.ok(proxy);
+
+      // Test proxied selector method.
+      assert.deepEqual(proxy.getThing('foo'), { id: 'foo' });
+      assert.deepEqual(proxy.getThing('bar'), { id: 'bar' });
+
+      // Test proxied action dispatch.
+      proxy.addThing('baz');
+      assert.calledWith(addThingSpy, 'baz');
+    });
+
+    it('proxies non-function properties of the store', () => {
```

Comment:
Are there any known current use cases for this, or are you merely illustrating that such a thing _would_ be possible? Should this be commented? Should it (access to non-function properties) be allowed?

Reply:
We don't currently have a specific need to be able to access non-function properties, so this is just for completeness.